### ISSUE TITLE
Unable to unset backward-compatible-eol in HttpSignProvider

### ISSUE THREAD

**Issue by janotav** [oracle/helidon#3907]
[//]: # "Please remove these comments"
[//]: # "Provide a general summary of the issue in the Title above"

## Environment Details
* Helidon Version: 2.4.2
* Helidon SE
* JDK version: 14.0.2
* OS: Linux
* Docker version (if applicable):

----------

## Problem Description
Unable to perform OCI signed requests. Due to typo in **OutboundTargetDefinition** the **backward-comptabible-eol** is always enabled even when config explicitly disables it.
```
 public Builder config(Config config) {
            Builder builder = new Builder();

            // mandatory
            builder.keyId(config.get("key-id").asString().get());
            config.get("header").asString().map(HttpSignHeader::valueOf).ifPresent(builder::header);
            config.get("sign-headers").as(SignedHeadersConfig::create).ifPresent(builder::signedHeaders);
            config.get("private-key").as(KeyConfig::create).ifPresent(builder::privateKeyConfig);
            config.get("hmac.secret").asString().ifPresent(builder::hmacSecret);

            // last, as we configure defaults based on configuration
            config.get("algorithm").asString().ifPresent(builder::algorithm);

            // backward compatibility with previous Helidon versions
            config.get("backward-compatible-eol").asBoolean().ifPresent(this::backwardCompatibleEol); // <--- should be builder::backwardCompatibleEol instead of this::backwardCompatibleEol, correct?

            return builder;
        }
```

**Comment by tomas-langer** [comment]
That is indeed a bug. The method should update the current instance, never create a new one.

**Comment by tomas-langer** [comment]
What kind of OCI requests are you doing? We do have a separate framework for calling OCI that works (it uses a dedicated OCI specific security provider). 
We have implementations for vault, object storage and a few other APIs for Helidon SE. 
There is more to the OCI communication than the signatures, and we should be covering that.

**Comment by janotav** [comment]
In the bigger picture, we need the ability to call various OCI services. Our primary scenario is calling OCI Datascience Model Deployment service using Resource Principals (2.2). Unfortunately this functionality is not implemented as of Helidon SE 2.4.2

**Comment by spericas** [comment]
PR #3913
